core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -581,9 +586,7 @@ class ReplicaManager(val config: KafkaConfig,
             FetchDataInfo(LogOffsetMetadata.UnknownOffsetMetadata, MemoryRecords.EMPTY)
         }
 
-        val readToEndOfLog = initialLogEndOffset.messageOffset - logReadInfo.fetchOffsetMetadata.messageOffset <= 0
-
-        LogReadResult(logReadInfo, localReplica.highWatermark.messageOffset, partitionFetchSize, readToEndOfLog, None)
+        LogReadResult(logReadInfo, initialHighWatermark, initialLogEndOffset, fetchTimeMs, partitionFetchSize, None)
```

Comment:
I am not sure if we need named params here because the variable name is already very close to the parameter name and should explain what they stand for. I can update it to use named parameters as well. Please let me know.